src/main/rx/jetiexbus.c

```diff
@@ -0,0 +1,595 @@
+    now = millis();
+
+    jetiExBusTimeInterval = now - jetiExBusTimeLast;
+    jetiExBusTimeLast = now;
+
+    if (jetiExBusTimeInterval > JETIEXBUS_MIN_FRAME_GAP) {
+        jetiExBusFrameReset();
+        jetiExBusFrameState = EXBUS_STATE_ZERO;
+        jetiExBusRequestState = EXBUS_STATE_ZERO;
+    }
+    
+    // Check if we shall start a frame?
+    if (jetiExBusFramePosition == 0) {
+        switch(c){
+        case JETIEXBUS_START_CHANNEL_FRAME:
+            jetiExBusFrameState = EXBUS_STATE_IN_PROGRESS;
+            jetiExBusFrame = &jetiExBusSerialFrame[0];
+            break;
+            
+        case JETIEXBUS_START_REQUEST_FRAME:
+            jetiExBusRequestState = EXBUS_STATE_IN_PROGRESS;
+            jetiExBusFrame = &jetiExBusSerialFrame[JETIEXBUS_MAX_FRAME_SIZE];
+            break;
+
+        default:
+            return;
+        }
+        crc = 0;
+    }
+
+    // Store in frame copy
+    jetiExBusFrame[jetiExBusFramePosition] = (uint8_t)c;
+    crc = updateCRC16(crc, (uint8_t)c);
+    jetiExBusFramePosition++;
+    
+    // Check the header for the message length
+    if (jetiExBusFramePosition==JETIEXBUS_HEADER_LEN) {
+        if ((jetiExBusFrame[EXBUS_HEADER_MSG_LEN]<=JETIEXBUS_MAX_FRAME_SIZE) && (jetiExBusFrame[EXBUS_HEADER_MSG_LEN]>=JETIEXBUS_MIN_FRAME_SIZE)) {
+            jetiExBusFrameLength = jetiExBusFrame[EXBUS_HEADER_MSG_LEN];
+        } else {
+            jetiExBusFrameReset();                  // not a valid frame
+            jetiExBusFrameState = EXBUS_STATE_ZERO;
+            jetiExBusRequestState = EXBUS_STATE_ZERO;
+            return;
+        }
+    }
+
+    // Done?
+    if (jetiExBusFrameLength == jetiExBusFramePosition) {
+        if (crc == 0){
+            if (jetiExBusFrameState == EXBUS_STATE_IN_PROGRESS) jetiExBusFrameState = EXBUS_STATE_RECEIVED;
+            if (jetiExBusRequestState == EXBUS_STATE_IN_PROGRESS) jetiExBusRequestState =  EXBUS_STATE_RECEIVED;
+        }
+        jetiExBusFrameReset();
+    }
+}
+
+
+// Check if it is time to read a frame from the data...
+uint8_t jetiExBusFrameStatus(void)
+{
+    if (jetiExBusFrameState == EXBUS_STATE_RECEIVED) {
+        jetiExBusDecodeFrame(&jetiExBusSerialFrame[0]);
+        jetiExBusFrameState = EXBUS_STATE_ZERO;
+    } else {
+        return SERIAL_RX_FRAME_PENDING;
+    }
+    return SERIAL_RX_FRAME_COMPLETE;
+}
+
+
+static uint16_t jetiExBusReadRawRC(rxRuntimeConfig_t *rxRuntimeConfig, uint8_t chan)
+{
+    if (chan >= rxRuntimeConfig->channelCount)
+        return 0;
+
+    return (jetiExBusChannelData[chan]);
+}
+
+#ifdef TELEMETRY
+/*
+  -----------------------------------------------
+   Jeti Ex Bus Telemetry
+  -----------------------------------------------
+*/
+void createExTelemetrieTextMessage(uint8_t *exMessage, uint8_t messageID, const exBusSensor_t *Sensor)
+{
+    uint8_t labelLength = strlen(Sensor->name);
+    uint8_t unitLength = strlen(Sensor->unit);
+
+    jetiExTelemetry[EXTEL_HEADER_LSN_LB] = messageID & 0xF0;
```

Comment:
sometimes its easier to gett pointer to buffer and store message byte-by-byte
```
uint8_t *p = exMessage;
*p++ =  8 + labelLength + unitLength;
*p++ = messageID & 0x0F; 
...
memcpy(p, Sensor->name, labelLength); p += labelLength;
...
```

You can use P to calculate actual length (p-jetiExTelemetry), but that's not that useful with CRC and length far apart ...